optaplanner-core/src/main/java/org/optaplanner/core/impl/score/buildin/hardsoft/HardSoftScoreInliner.java

```diff
@@ -16,61 +16,59 @@
 
 package org.optaplanner.core.impl.score.buildin.hardsoft;
 
-import java.util.function.Consumer;
+import java.util.List;
+import java.util.function.Supplier;
 
-import org.optaplanner.core.api.score.Score;
 import org.optaplanner.core.api.score.buildin.hardsoft.HardSoftScore;
 import org.optaplanner.core.impl.score.inliner.IntWeightedScoreImpacter;
 import org.optaplanner.core.impl.score.inliner.ScoreInliner;
 
 public class HardSoftScoreInliner extends ScoreInliner<HardSoftScore> {
 
-    protected int hardScore;
-    protected int softScore;
+    private int hardScore;
+    private int softScore;
 
     protected HardSoftScoreInliner(boolean constraintMatchEnabled) {
-        super(constraintMatchEnabled);
+        super(constraintMatchEnabled, HardSoftScore.ZERO);
     }
 
     @Override
-    public IntWeightedScoreImpacter buildWeightedScoreImpacter(HardSoftScore constraintWeight) {
-        if (constraintWeight.equals(HardSoftScore.ZERO)) {
-            throw new IllegalArgumentException("The constraintWeight (" + constraintWeight + ") cannot be zero,"
-                    + " this constraint should have been culled during node creation.");
-        }
+    public IntWeightedScoreImpacter buildWeightedScoreImpacter(String constraintPackage, String constraintName,
+            HardSoftScore constraintWeight) {
+        ensureNonZeroConstraintWeight(constraintWeight);
         int hardConstraintWeight = constraintWeight.getHardScore();
         int softConstraintWeight = constraintWeight.getSoftScore();
         if (softConstraintWeight == 0) {
-            return (int matchWeight, Consumer<Score<?>> matchScoreConsumer) -> {
+            return (int matchWeight, Supplier<List<Object>> justifications) -> {
```

Comment:
Naming: justifications isn't an array or Collection of justification objects. It's a supplier that returns that list.
=> Rename justifications to justificationsSupplier (across entire PR)